packages/pages/src/common/src/template/client.ts

```diff
@@ -44,22 +43,25 @@ export const makeClientFiles = async (projectStructure: ProjectStructure) => {
  * @param path src/templates/<templateName>.tsx
  */
 const generateAndSaveClientHydrationTemplates = async (path: string) => {
+  const newPath = getClientPath(path);
+  fs.writeFileSync(newPath, "");
   const sfp = new SourceFileParser(path, createTsMorphProject());
-  const newSfp = new SourceFileParser(
-    formatClientPath(path),
-    createTsMorphProject()
-  );
+  const newSfp = new SourceFileParser(newPath, createTsMorphProject());
   const templateParser = new TemplateParser(sfp).makeClientTemplateFromSfp(
     newSfp
   );
   templateParser.sourceFile.save();
 };
 
-const formatClientPath = (clientPath: string): string => {
-  const templatePath = path.parse(clientPath);
+/**
+ * @param templatePath /src/templates/location.tsx
+ * @returns clientPath /src/templates/location.client.tsx
+ */
+const getClientPath = (templatePath: string): string => {
+  const parsedPath = path.parse(templatePath);
   return path.join(
-    templatePath.dir,
-    templatePath.name + ".client" + templatePath.ext
+    parsedPath.dir,
```

Comment:
I think you can just update the parsedPath.name and then call path.format(parsedPath)

Reply:
Done. Had to update parsedPath.base as well for format to work